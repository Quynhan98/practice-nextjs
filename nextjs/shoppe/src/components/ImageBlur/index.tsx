import Image, { ImageProps } from 'next/image'

// Services
// import imageLoader from '@services/imageLoader'

// Utils
// import { shimmer, toBase64 } from '@utils/index'
import { memo } from 'react'

const ImageBlur = ({ width, height, ...rest }: ImageProps) => {
  return (
    <Image
      width={width}
      height={height}
      // loader={imageLoader as ImageLoader}
      // placeholder="blur"
      // blurDataURL={`data:image/svg+xml;base64,${toBase64(
      //   shimmer(width as number, height as number),
      // )}`}
      {...rest}
    />
  )
}

export default memo(ImageBlur)
