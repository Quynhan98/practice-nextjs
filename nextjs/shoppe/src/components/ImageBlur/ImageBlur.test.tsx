import React from 'react'

// Component
import ImageBlur from '@components/ImageBlur'

// Utils
import { render, screen } from '@utils/testUtils'

describe('Image render', () => {
  const props = {
    src: '/facebook.webp',
    alt: 'facebook',
    width: 200,
    height: 200,
  }
  it('Should show match Image DOM Snapshot', () => {
    const { container } = render(<ImageBlur {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('Should image has the correct src', () => {
    render(<ImageBlur {...props} />)

    const image = screen.getByRole('img')

    expect(image.getAttribute('src')).toEqual(
      `${process.env.NEXT_PUBLIC_BASE_URL}/${props.src}?w=640&q=75`,
    )
  })
})
