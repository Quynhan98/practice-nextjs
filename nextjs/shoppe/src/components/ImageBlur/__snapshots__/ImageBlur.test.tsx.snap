// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image render Should show match Image DOM Snapshot 1`] = `
<div>
  <img
    alt="facebook"
    data-nimg="1"
    decoding="async"
    height="200"
    loading="lazy"
    src="http://localhost:3000//facebook.webp?w=640&q=75"
    srcset="http://localhost:3000//facebook.webp?w=256&q=75 1x, http://localhost:3000//facebook.webp?w=640&q=75 2x"
    style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
    width="200"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
